### PR TITLE
0.6.5 China fixes

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
These changes fix several issues:

- Spatial CLI for China was changes and the "domain" flag was deprecated. The flag was removed from one place in the code and from two setup scripts. 
- An issue in cloud deployment where the upload snapshot http request was sent with a missing header in the case of China.
- The DeploymentLauncher was using the west refresh token when deploying to China.
- We previously relied on Spot to run the auth process for us when launching locally. The Spot process ends before finishing the auth process in the case of China. This meant we would continue with out deployment without being authed and run into errors. We know explicitly run the auth process on cloud and local deployment to ensure this does not happen. This does not seem to incur any significant time cost as attempting to auth whilst already authed will quickly finish.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp [hunk 1]
```diff
@@ -505,38 +509,62 @@ FReply SSpatialGDKSimulatedPlayerDeployment::OnLaunchClicked()
...
+				FSimpleDelegate::CreateLambda([NotificationItem]() {
 				NotificationItem->SetText(FText::FromString(TEXT("Failed to launch the DeploymentLauncher script properly.")));
 				NotificationItem->SetCompletionState(SNotificationItem::CS_Fail);
 				NotificationItem->SetExpireDuration(7.5f);
 				NotificationItem->ExpireAndFadeout();
 			})
-		);
-		return FReply::Handled();
-	}
+				);
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Nit: Whitespace seems a bit weird here, may just be GitHub.

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp [hunk 2]
```diff
@@ -505,38 +509,62 @@ FReply SSpatialGDKSimulatedPlayerDeployment::OnLaunchClicked()
...
+	TSharedPtr<SNotificationItem> NotificationItem = FSlateNotificationManager::Get().AddNotification(Info);
 
-		NotificationItem->SetCompletionState(SNotificationItem::CS_Pending);
+	NotificationItem->SetCompletionState(SNotificationItem::CS_Pending);
 
-		SpatialGDKEditorSharedPtr->LaunchCloudDeployment(
-			FSimpleDelegate::CreateLambda([NotificationItem]() {
+	auto LaunchCloudDeployment = [this, NotificationItem]()
+	{
+		if (TSharedPtr<FSpatialGDKEditor> SpatialGDKEditorSharedPtr = SpatialGDKEditorPtr.Pin()) {
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]:
```suggestion
		if (TSharedPtr<FSpatialGDKEditor> SpatialGDKEditorSharedPtr = SpatialGDKEditorPtr.Pin())
		{
```

File: SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	FPlatformProcess::ExecProcess(*SpatialExe, *SpatialInfoArgs, &ExitCode, &SpatialInfoResult, &StdErr);
+
+	bool bSuccess = ExitCode == 0;
+	if (!bSuccess)
+	{
+		UE_LOG(LogSpatialCommandUtils, Warning, TEXT("Spatial auth login failed. Error Code: %d, Error Message: %s"), ExitCode, *SpatialInfoResult);
+	}
+
+	return bSuccess;
+}
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp]: Nit: Newline to make GitHub not show the annoying icon?

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 4]
```diff
@@ -311,7 +285,58 @@ bool FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
...
+
+	// If the service is not running then start it.
+	if (!bSpatialServiceRunning)
+	{
+		TryStartSpatialService();
+	}
+
+#if ENGINE_MINOR_VERSION <= 22
+	AttemptSpatialAuthResult = Async<bool>(EAsyncExecution::Thread, [this]() { return SpatialCommandUtils::AttemptSpatialAuth(bIsInChina); },
+#else
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]: 0.6 does not support >4.23, so we could theoretically take out this guard, also I am slightly worried whether this works on 4.20 (which 0.6 should).

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp [hunk 5]
```diff
@@ -9,7 +9,11 @@
 #include "Framework/MultiBox/MultiBoxBuilder.h"
 #include "Framework/Notifications/NotificationManager.h"
 #include "Templates/SharedPointer.h"
+#include "SpatialCommandUtils.h"
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Sort these includes

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp [hunk 6]
```diff
@@ -505,38 +509,62 @@ FReply SSpatialGDKSimulatedPlayerDeployment::OnLaunchClicked()
...
+		Info.bUseSuccessFailIcons = true;
+		Info.ExpireDuration = 3.0f;
 
-	TSharedPtr<SNotificationItem> NotificationItem = FSlateNotificationManager::Get().AddNotification(Info);
-	NotificationItem->SetCompletionState(SNotificationItem::CS_Fail);
+		TSharedPtr<SNotificationItem> NotificationItem = FSlateNotificationManager::Get().AddNotification(Info);
+		NotificationItem->SetCompletionState(SNotificationItem::CS_Fail);
+	};
+
+#if ENGINE_MINOR_VERSION <= 22
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Removed all 4.23 references

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 7]
```diff
@@ -16,6 +16,7 @@
 #include "Serialization/JsonSerializer.h"
 #include "Serialization/JsonWriter.h"
 #include "SpatialGDKServicesModule.h"
+#include "SpatialCommandUtils.h"
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]: Sort please

File: SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+bool SpatialCommandUtils::AttemptSpatialAuth(bool IsRunningInChina)
+{
+	FString SpatialInfoArgs = IsRunningInChina ? TEXT("auth login --environment=cn-production") : TEXT("auth login");
+	FString SpatialInfoResult;
+	FString StdErr;
+	int32 ExitCode;
+
+	const static FString SpatialExe = TEXT("spatial");
+	FPlatformProcess::ExecProcess(*SpatialExe, *SpatialInfoArgs, &ExitCode, &SpatialInfoResult, &StdErr);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp]: Do we have access to `SpatialGDKServicesConstants::SpatialExe`?
Or is that not in this release.
If we don't - please pass TEXT("spatial") directly.

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp]: In 0.6 we don't have this in a nice place. We do in 0.8 though.

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp (hunk 2 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Missed this

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 9]
```diff
@@ -311,7 +285,58 @@ bool FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
...
+
+	bStartingDeployment = true;
+
+	// If the service is not running then start it.
+	if (!bSpatialServiceRunning)
+	{
+		TryStartSpatialService();
+	}
+
+#if ENGINE_MINOR_VERSION <= 22
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]: 423 support here

File: SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp (hunk 8 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp]: Can you pass TEXT("spatial") directly? No need for a static.

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp [hunk 10]
```diff
@@ -505,38 +508,65 @@ FReply SSpatialGDKSimulatedPlayerDeployment::OnLaunchClicked()
...
-		Info.bFireAndForget = false;
+	FNotificationInfo Info(FText::FromString(TEXT("Starting simulated player deployment...")));
+	Info.bUseSuccessFailIcons = true;
+	Info.bFireAndForget = false;
 
-		TSharedPtr<SNotificationItem> NotificationItem = FSlateNotificationManager::Get().AddNotification(Info);
+	TWeakPtr<SNotificationItem> NotificationItem = FSlateNotificationManager::Get().AddNotification(Info);
 
-		NotificationItem->SetCompletionState(SNotificationItem::CS_Pending);
+	NotificationItem.Pin()->SetCompletionState(SNotificationItem::CS_Pending);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKSimulatedPlayerDeployment.cpp]: Pin can fail - you should validate it before using. I realise that's very unlikely as the notification was just made, but worth doing.

**Pull request merged**